feat: Add click functionality to banner carousel images

OTAKUWeBer opened the pull request:
## Changes
- Added clickable functionality to banner carousel images
- Banner images now show pointer cursor on hover
- Click handler triggers navigation when tapping on banner image
- Text content (title, rating, description) remains non-interactive

## Technical Details
- Wrapped banner image (`ClipRRect`) with `MouseRegion` for cursor styling
- Added `GestureDetector` to handle tap events
- Clickable area is limited to the image container only
- Text elements below remain outside the interactive region

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Banners on anime and manga screens are interactive — tapping a banner opens the item's detail view and shows a pointer on hover; the selected item and its surrounding list are forwarded to details.
  * Complete video player controls UI — layered overlays, persistent auto-hide, prominent play/pause, improved seek/drag UX, volume/speed displays, and skip overlay.

* **Enhancements**
  * Keyboard shortcuts (play/pause, seek, volume, mute, fullscreen, speed), mute/fullscreen toggles, and safer volume/playback speed handling.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

OTAKUWeBer committed:
feat: make banner carousel image clickable

- Add MouseRegion with click cursor to banner image
- Add GestureDetector to handle tap events on banner
- Clickable area limited to image only, not text content

OTAKUWeBer committed:
Change for pubsec.yaml

OTAKUWeBer committed:
fix(banner_carousel): add assertion and fix text/image mismatch

- Add constructor assertion for animeList/mangaList
- Fix title/rating/description using wrong index (_currentPage vs index)

OTAKUWeBer committed:
Merge branch 'K3vinb5:rewrite' into rewrite

OTAKUWeBer committed:
feat: pass MediaList to UnyoBannerCarousel for navigation context

OTAKUWeBer committed:
feat: pass MediaList to UnyoBannerCarousel for navigation context

OTAKUWeBer committed:
feat: Discord version update

K3vinb5 commented on lib/presentation/screens/manga_screen.dart:
```diff
@@ -119,7 +128,7 @@ class _MangaViewState extends State<_MangaView> {
                     mangaList: state.trending.$2,
                     controller: trendingController,
-                    onPressed: context.read<MangaCubit>().navigateToMangaDetails,
+                    onPressed: (manga, mediaList) => context.read<MangaCubit>().navigateToMangaDetails(manga, mediaList),
```
remove this arguments, just pass the reference to the function, same for the others. Do it the same you did in the anime_screen, no reason to do it differently

OTAKUWeBer replied:
done

K3vinb5 commented on lib/presentation/screens/manga_screen.dart:
```diff
@@ -129,7 +138,7 @@ class _MangaViewState extends State<_MangaView> {
                     mangaList: state.recentlyCompleted.$2,
                     controller: recentlyCompletedController,
-                    onPressed: context.read<MangaCubit>().navigateToMangaDetails,
+                    onPressed: (manga, mediaList) => context.read<MangaCubit>().navigateToMangaDetails(manga, mediaList),
```
Same here and for the others in this file btw

OTAKUWeBer replied:
done

K3vinb5 commented on lib/presentation/widgets/styled/unyo_banner_carousel.dart:
I have reviewed this file and basically you did too much and overcomplicated things, 2 lines would achieve the same

K3vinb5 replied:
Just Wrap the ClipRRect with an Inkwell and pass the same border radius to the Inkwell, then plug the method onPressed you're receiving into onTap from the InkWell. I please ask you to not use chatgpt for these stuff, to do good flutter code you need to have some basics from flutter

OTAKUWeBer replied:
Now check .

OTAKUWeBer committed:
Merge branch 'K3vinb5:rewrite' into rewrite

OTAKUWeBer committed:
Make it simple

OTAKUWeBer committed:
pass the reference to the function

OTAKUWeBer committed:
Merge branch 'K3vinb5:rewrite' into rewrite

OTAKUWeBer committed:
Merge branch 'K3vinb5:rewrite' into rewrite

OTAKUWeBer committed:
Merge branch 'K3vinb5:rewrite' into rewrite

OTAKUWeBer committed:
fix(video): resolve play/pause and slider issues in video player

- Remove buffering check blocking play/pause in VideoService
- Add local isPlaying state tracking with optimistic updates
- Fix slider glitches with proper drag state management and value clamping
- Add formatDuration helper for consistent time display
- Add ESC key handler to exit video player
- Improve UI with animated play/pause transitions and volume slider
- Add playback speed indicator in bottom controls
